dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,151 @@
-    private readonly string _apiKey;
-
-    /// <summary>
-    /// Maximum number of attempts to retrieve the text to image operation result.
-    /// </summary>
-    private readonly int _maxRetryCount;
-
-    /// <summary>
-    /// Azure OpenAI Endpoint ApiVersion
-    /// </summary>
-    private readonly string _apiVersion;
-
+    private readonly OpenAIClient _client;
+    private readonly ILogger _logger;
+    private readonly string _deploymentName;
     /// <summary>
-    /// Create a new instance of Azure OpenAI text to image service
+    /// Create a new instance of Azure OpenAI image generation service
     /// </summary>
-    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="deploymentName">Deployment name identifier</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL</param>
+    /// <param name="apiKey">Azure OpenAI API key</param>
+    /// <param name="modelId">Model name identifier</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
-    /// <param name="maxRetryCount"> Maximum number of attempts to retrieve the text to image operation result.</param>
     /// <param name="apiVersion">Azure OpenAI Endpoint ApiVersion</param>
     public AzureOpenAITextToImageService(
-        string? endpoint,
+        string deploymentName,
+        string endpoint,
         string apiKey,
         string? modelId = null,
         HttpClient? httpClient = null,
         ILoggerFactory? loggerFactory = null,
-        int? maxRetryCount = null,
         string? apiVersion = null)
     {
         Verify.NotNullOrWhiteSpace(apiKey);
+        Verify.NotNullOrWhiteSpace(deploymentName);
+
+        this._deploymentName = deploymentName;
 
-        if (httpClient?.BaseAddress == null && string.IsNullOrEmpty(endpoint))
+        this._logger = loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)) ?? NullLogger.Instance;
+        if (string.Equals(modelId, "dall-e-2", StringComparison.OrdinalIgnoreCase))
         {
-            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+            this._logger.LogWarning($"{nameof(AzureOpenAITextToImageService)} supports only Dall-E-3. .");
+            throw new NotSupportedException("Dall-E-2 support was deprecated in Azure Open AI latest SDK. Please use a 'dall-e-3' deployment.");
```

Comment:
I still think that this condition is not required because `modelId` property is not used in this class. Whichever `modelId` is passed, it doesn't impact the functionality, which means that we should not throw an exception and probably `modelId` property can be removed from constructor. In case if developer uses Azure deployment, that is not supported, Azure SDK will throw appropriate exception and we will pass it to the user.

Reply:
Removed